extensions/cba_settings/cba_settings.cpp

```diff
@@ -0,0 +1,335 @@
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        line = trim_left(line);
+
+        // Ignore commented out lines.
+        if (line.substr(0, 2) == "//")
+            continue;
+
+        // Count force tags.
+        int force = 0;
+        while (line.substr(0, 6) == "force ") {
+            force++;
+            line = line.substr(6);
+            line = trim_left(line);
+        }
+
+        // Get setting.
+        size_t i = line.find("=");
+        if (i == string::npos) // Ignore ill-formed line.
+            continue;
+
+        string item = line.substr(0, i);
+        item = trim_right(item);
+
+        // Get value.
+        string value = line.substr(i + 1);
+        value = trim(value);
+
+        while (value.back() == ';')
+            value.pop_back();
+
+        value = trim_right(value);
+
+        result = result + "[\"" + item + "\"," + value + "," + to_string(force) + "],";
+    }
+
+    file.close();
+
+    result.pop_back();
+    result.push_back(']');
+
+    if (strlen * MAX_CALLBACK_LENGTH > result.length())
+        return "";
+
+    result = result.substr(strlen * MAX_CALLBACK_LENGTH, MAX_CALLBACK_LENGTH);
+    return result;
+}
+
+string read(string setting) {
+    string result = "[\"\",false,-1]";
+
+    ifstream file;
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        line = trim_left(line);
+
+        // Ignore commented out lines.
+        if (line.substr(0, 2) == "//")
+            continue;
+
+        // Count force tags.
+        int force = 0;
+        while (line.substr(0, 6) == "force ") {
+            force++;
+            line = line.substr(6);
+            line = trim_left(line);
+        }
+
+        // Get setting.
+        size_t i = line.find("=");
+        if (i == string::npos) // Ignore ill-formed line.
+            continue;
+
+        string item = line.substr(0, i);
+        item = trim_right(item);
+
+        if (_stricmp(item.c_str(), setting.c_str()) == 0) {
+            // Get value.
+            string value = line.substr(i + 1);
+            value = trim(value);
+
+            while (value.back() == ';')
+                value.pop_back();
+
+            value = trim_right(value);
+
+            result = "[\"" + item + "\"," + value + "," + to_string(force) + "]";
+            break;
+        }
+    }
+
+    file.close();
+    return result;
+}
+
+string append(string setting, string value, int force) {
+    ofstream file;
+    file.open(SETTINGS_FILE, ios::app);
+
+    string line = "";
+    while (force-- > 0)
+        line = line + "force ";
+
+    line = line + setting + " = " + value + ";";
+
+    file << line << endl;
+    file.close();
+    return "true";
+}
+
+string write(string setting, string value, int force) {
+    string result = "false";
+
+    ofstream temp;
+    temp.open(string(TEMP_FILE).c_str(), ios::out);
```

Comment:
you are reloading, reparsing, rewriting the whole file for every setting property write?
You could just parse it once, and store it in memory. Your extension won't get unloaded

Reply:
>store it in memory
How?

Reply:
Comment below, just a global variable. Array or map or some container